jgiven-core/src/main/java/com/tngtech/jgiven/impl/inject/ValueInjector.java

```diff
@@ -90,6 +90,8 @@ public void updateValues( Object object ) {
                 if( value != null ) {
                     field.set( object, value );
                     log.debug( "Setting field {} to value {}", field, value );
+                } else if( isRequired( field ) ) {
```

Comment:
I think the answer to »can it« is a clear yes. I'm not sure about whether it's worth it, though. Fields generally don't have dozens, hundreds or thousands of annotations, so my first guess is that we're optimizing a non-issue here. Do you have specific concerns about performance for this?